CodeGeneration/Sources/SyntaxSupport/AvailabilityNodes.swift

```diff
@@ -118,42 +118,50 @@ public let AVAILABILITY_NODES: [Node] = [
     ]
   ),
 
-  // version-tuple -> integer-literal
-  //                | float-literal
-  //                | float-literal '.' integer-literal
+  // version-tuple-element -> '.' interger-literal
   Node(
-    name: "VersionTuple",
-    nameForDiagnostics: "version tuple",
-    description: "A version number of the form major.minor.patch in which the minor and patch part may be omitted.",
+    name: "VersionComponent",
+    nameForDiagnostics: nil,
+    description: "An element to represent a single component in a version, like `.1`.",
     kind: "Syntax",
     children: [
       Child(
-        name: "Major",
-        kind: .token(choices: [.token(tokenKind: "IntegerLiteralToken")]),
-        description: "The major version."
-      ),
-      Child(
-        name: "MinorPeriod",
+        name: "Period",
         kind: .token(choices: [.token(tokenKind: "PeriodToken")]),
-        description: "If the version contains a minor number, the period separating the major from the minor number.",
-        isOptional: true
+        description: "The period of this version component"
       ),
       Child(
-        name: "Minor",
+        name: "Number",
         kind: .token(choices: [.token(tokenKind: "IntegerLiteralToken")]),
-        description: "The minor version if specified.",
-        isOptional: true
+        description: "The version number of this component"
       ),
+    ]
+  ),
+
+  // version-list -> version-tuple-element version-list?
+  Node(
+    name: "VersionComponentList",
+    nameForDiagnostics: nil,
+    kind: "SyntaxCollection",
+    element: "VersionComponent",
+    omitWhenEmpty: true
+  ),
+
+  // version-tuple -> integer-literal version-list?
+  Node(
+    name: "VersionTuple",
+    nameForDiagnostics: "version tuple",
+    description: "A version number of the form major.minor.patch in which the minor and patch part may be omitted.",
+    kind: "Syntax",
+    children: [
       Child(
-        name: "PatchPeriod",
-        kind: .token(choices: [.token(tokenKind: "PeriodToken")]),
-        description: "If the version contains a patch number, the period separating the minor from the patch number.",
-        isOptional: true
+        name: "Major",
+        kind: .token(choices: [.token(tokenKind: "IntegerLiteralToken")]),
+        description: "The major version."
       ),
       Child(
-        name: "Patch",
-        kind: .token(choices: [.token(tokenKind: "IntegerLiteralToken")]),
-        description: "The patch version if specified.",
+        name: "Components",
+        kind: .collection(kind: "VersionComponentList", collectionElementName: "VersionComponent"),
```

Comment:
Again, sorry for not catching it earlier. I’m trying to generally improve documentation of syntax nodes, it would be nice to add one here. I’d suggest

```swift
description: "Any version components that are not the major version . For example, for `1.2.0`, this will contain `.2.0`"
```